### PR TITLE
Add rest of 2 year jetpack plan constants

### PR DESCRIPTION
**Pull request by CodeyGuyDylan**
## Proposed Changes

* Add new 2 year plan constants wherever else they are needed
* Most of the places added in this PR are in tests

## Testing Instructions

1. Make sure all tests pass
2. In your 0-sandbox.php file, add the following line
```php
define( 'USE_STORE_SANDBOX', true );
```
3. Sandbox `public-api.wordpress.com`
4. Using the Calypso live link or your local environment with this branch checked out, make sure you can still purchase the 2 year plans with no issues

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

**Comment by CodeyGuyDylan** [comment]
This is built off https://github.com/Automattic/wp-calypso/pull/81148it will be rebased when that one is approved

**Review by robertsreberski — APPROVED** [review]
Looks solid 💪 I confirm I was able to purchase plans, tests are all passing, code looks good.

File: packages/calypso-products/test/is-jetpack-legacy-item.js [hunk 1]
```diff
@@ -15,8 +17,10 @@ describe( 'isJetpackLegacyItem', () => {
 	} );
 
 	it( 'should return false otherwise', () => {
+		expect( isJetpackLegacyItem( PRODUCT_JETPACK_ANTI_SPAM_BI_YEARLY ) ).toEqual( false );
 		expect( isJetpackLegacyItem( PRODUCT_JETPACK_ANTI_SPAM ) ).toEqual( false );
 		expect( isJetpackLegacyItem( PRODUCT_JETPACK_ANTI_SPAM_MONTHLY ) ).toEqual( false );
+		expect( isJetpackLegacyItem( PLAN_JETPACK_COMPLETE_BI_YEARLY ) ).toEqual( false );
```
Comment (ddc22) [inline packages/calypso-products/test/is-jetpack-legacy-item.js:23]: Nothing critical (And I know this change is unrelated), but there might be some better words to use for multi year plans :)

https://github.com/Automattic/wp-calypso/blob/370b3b8ef2426b65732a81fc4a5d68dae257df6c/packages/calypso-products/src/constants/terms.ts#L27-L37

Comment (CodeyGuyDylan) [reply packages/calypso-products/test/is-jetpack-legacy-item.js:23]: In this case I was using the same naming convention set by the previous Jetpack 2 year plans `JETPACK_SECURITY_T1_BI_YEARLY` and `JETPACK_BACKUP_T1_BI_YEARLY`

Thank you for pointing this out though and I will keep it in mind in the future 😄

**Pull request merged**